src/System.Management.Automation/engine/SessionStateProviderAPIs.cs

```diff
@@ -1510,285 +1452,6 @@ private void NewProviderEntry(ProviderInfo provider)
-                    {
-                        if (force)
-                        {
-                            // Forcefully remove all the drives
-
-                            foreach (PSDriveInfo drive in GetDrivesForProvider(providerName))
-                            {
-                                if (drive != null)
-                                {
-                                    RemoveDrive(drive, true, null);
-                                }
-                            }
-                        }
-                        else
-                        {
-                            errors = true;
-
-                            // Since there are still drives associated with the provider
-                            // the provider cannot be removed
-
-                            SessionStateException e = new SessionStateException(
-                                providerName,
-                                SessionStateCategory.CmdletProvider,
-                                "RemoveDrivesBeforeRemovingProvider",
-                                SessionStateStrings.RemoveDrivesBeforeRemovingProvider,
-                                ErrorCategory.InvalidOperation);
-                            context.WriteError(new ErrorRecord(e.ErrorRecord, e));
-
-                            return;
-                        }
-                    }
-
-                    // Now tell the provider that they are going to be removed by
-                    // calling the Stop method
-
-                    try
-                    {
-                        providerBase.Stop(context);
-                    }
-                    catch (LoopFlowException)
-                    {
-                        throw;
-                    }
-                    catch (PipelineStoppedException)
-                    {
-                        throw;
-                    }
-                    catch (ActionPreferenceStopException)
-                    {
-                        throw;
-                    }
-                }
-            }
-            catch (LoopFlowException)
-            {
-                throw;
-            }
-            catch (PipelineStoppedException)
-            {
-                throw;
-            }
-            catch (ActionPreferenceStopException)
-            {
-                throw;
-            }
-            catch (Exception e)
-            {
-                errors = true;
-                context.WriteError(
-                    new ErrorRecord(
-                        e,
-                        "RemoveProviderUnexpectedException",
-                        ErrorCategory.InvalidArgument,
-                        providerName));
-            }
-            finally
-            {
-                if (force || !errors)
-                {
-                    // Log the provider stopped event
-
-                    MshLog.LogProviderLifecycleEvent(
-                        this.ExecutionContext,
```

Comment:
It's possible this event is used by forensic tools - as much as I'd like to get rid of it.

Reply:
If we know that RemoveProvider is used outside of RunspaceConfig, I can put this back, but currently it doesn't seem to be the case.